docs/scos/dev/keeping-a-project-upgradable/upgradability-guidelines/dependency-provider-contains-addition-logic.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Dependency provider contains addition logic
+description: Reference information for evaluator tools.
+template: howto-guide-template
+---
+
+Additional logic inside the dependency provider’s methods.
+
+## Problem description
+
+On the project level, developers use IF constructs with variety of expressions in dependency providers to register the plugins in particular cases only.
```

Comment:
IF -> `if`
Please, use '`' for the code fencing. The same for the rest IFs in this page.

Reply:
++